third_party/move/move-compiler-v2/src/file_format_generator/function_generator.rs

```diff
@@ -897,6 +905,51 @@ impl<'a> FunctionGenerator<'a> {
         self.code.push(bc)
     }
 
+    /// Prepare a temporary, i.e., attempt to put it on the stack.
+    /// It is always safe to consider this instruction as a no-op.
+    fn prepare_arg(&mut self, ctx: &BytecodeContext, temp: TempIndex) {
+        let fun_ctx = ctx.fun_ctx;
+        let PrepareUseAnnotation(prepare_use_map) = fun_ctx
+            .fun
+            .get_annotations()
+            .get::<PrepareUseAnnotation>()
+            .expect("prepare-use annotation is a prerequisite");
+        let use_info = prepare_use_map
+            .get(&ctx.code_offset)
+            .expect("code offset for `Prepare` must be in the map");
+        // If it is already on the stack, let's not mess with the stack. Make this a no-op.
+        if self.stack.iter().any(|(t, _)| *t == temp) {
+            return;
+        }
+        // If this arg appears multiple times in the use instruction, make this a no-op.
+        let use_instr = &fun_ctx.fun.data.code[use_info.offset as usize];
+        if use_instr.sources().iter().filter(|s| *s == &temp).count() > 1 {
+            return;
+        }
+        let live_var_annotation = fun_ctx
+            .fun
+            .get_annotations()
+            .get::<LiveVarAnnotation>()
+            .expect("livevar analysis is a prerequisite");
+        let alive_after = !use_instr.dests().contains(&temp)
+            && live_var_annotation
+                .get_live_var_info_at(use_info.offset)
+                .map(|a| a.after.contains_key(&temp))
+                .unwrap_or(false);
+        let local: u8 = self.temp_to_local(fun_ctx, Some(ctx.attr_id), temp);
```

Comment:
If you want to give it a type, perhaps not `u8` but `FF::LocalIndex` or what that is.